### DISCUSSION TITLE
File scanning on storage

### DISCUSSION THREAD

**Discussion by leegould** [supabase/supabase#23645] [Questions]
Hi,

I'm wondering if there is a good way to scan files uploaded to the storage api for viruses, malware etc. During a pen test recently, they were recommending we have a look at doing this. Anyone tried to achieve this yet?

Thanks for any help

**Comment by dragos199993** [comment]
Any updates on this one? Is there  something in place by default on Supabase Storage?

**Reply by GaryAustin1** [reply]
There is nothing in place.
You could do an edge function checking for new files and using edge code or 3rd party service to scan files (possibly using the s3 client connection).

**Reply by dragos199993** [reply]
I am aware this might not the right place to ask this, but do you have some recommendations for 3rd party service for this? 🙏 Thanks!

**Reply by GaryAustin1** [reply]
Sorry, not something I've actually done.  Just throwing out idea.

**Comment by Tommy1223454** [comment]
A practical pattern here is to keep storage as storage, and do the trust decision before or immediately around the storage boundary.

That’s the exact layer I built Pompelmi for in Node.js:
https://github.com/pompelmi/pompelmi

It’s not “Supabase-native”, but it works well as an app-side inspection gate for untrusted uploads before they’re treated as accepted. The useful part is that it goes beyond extension/client MIME checks and can feed reject / quarantine / promote workflows.
